update triton pin for release 2.3

shunting314 opened the pull request:
The current status of the PR:

- CI is clean. The only failures not related to building a wheel also happens on the base commit. 
- Perf test is mostly neutral [link](https://hud.pytorch.org/benchmark/compilers?startTime=Mon%2C%2011%20Mar%202024%2018%3A29%3A20%20GMT&stopTime=Mon%2C%2018%20Mar%202024%2018%3A29%3A20%20GMT&granularity=hour&suite=torchbench&mode=training&dtype=amp&lBranch=shunting-update-triton-release&lCommit=2f83d1c7345021a28639ba96dab5f0ac32a65540&rBranch=main&rCommit=9c50ecc84b9a6e699a7f058891b889aafbf976c7). The is an extra failed model in torchbench. But the error type 'eager_two_runs_differ' indicates it's not related to triton upgrade. We've double checked that the perf test job is using the correct triton pin (thanks to @huydhn  and @atalman )

I'll now clean up the PR to only keep the necessary PyTorch side change to make it work with tirton 2.2.x . Other hacks coming up to do the tests will be removed (e.g. pointing to my triton fork).

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @desertfire @chauhang

peterbell10 committed:
CI: Specify libc and libstdcxx versions in conda environments (#121556)

Without this we get mismatches between the GLIBC and GLIBCXX ABI used
by conda packages vs pytorch.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/121556
Approved by: https://github.com/isuruf, https://github.com/malfet

atalman commented on .ci/docker/ci_commit_pins/triton.txt:
```diff
@@ -1 +1 @@
-a9bc1a36470eefafe0e2ab2503b8698f1e89e7e3
+release/2.2.x
```
@shunting314 this will not work. Let me handle enabling 2.2 in CI. I would need to revert https://github.com/pytorch/pytorch/pull/121765 and builder PR first.

shunting314 committed:
update triton pin for release 2.3

shunting314 committed:
Revert "[Release Only] Build triton using pinned version rather branch (#121765)"

This reverts commit d69c4219127e2cf5d9637b0daacc0a24e65f8133.

shunting314 commented:
cc @atalman  for the error regarding building triton wheels

shunting314 commented:
We also see a bunch of GLIBCXX related failures. I'm worried that they may hide other triton related issues. @atalman

atalman committed:
Checkout release branch rather then commit_hash when building triton release (#115379) (#121901)

Cherry pick of https://github.com/pytorch/pytorch/pull/115379 from Release 2.2 that should be applied to main and Release 2.3 as well

Pull Request resolved: https://github.com/pytorch/pytorch/pull/121901
Approved by: https://github.com/DanilBaibak, https://github.com/jeanschmidt

peterbell10 commented:
GLIBCXX error is fixed by #121556

shunting314 committed:
Copy from https://github.com/pytorch/pytorch/pull/121916/commits/6e9373f0be6478710b2053132ffd5837896e05b3

shunting314 committed:
fix user defined triton kernel

shunting314 committed:
temporarily use my triton fork for testing

shunting314 commented:
CI is finally clean! The only failures not related to building a wheel also happens on the base commit. Will check how perf looks
